### ISSUE TITLE
Re-activate funlen checks but not in test

### ISSUE THREAD

**Issue by NicolasMahe** [mesg-foundation/engine#1332]
Since PR <https://github.com/mesg-foundation/engine/pull/1322>, golangci introduce a new check: funlen.
The codebase doesn't pass this check:
```
golangci-lint run00:12
Exit code: 1
#!/bin/bash -eo pipefail
golangci-lint run
hash/structhash/structhash.go:77: Function 'write' has too many statements (76 > 40) (funlen)
func write(buf *bytes.Buffer, v reflect.Value) *bytes.Buffer {
sdk/instance/start.go:11: Function 'start' is too long (89 > 60) (funlen)
func (i *Instance) start(inst *instance.Instance, imageHash string, env []string) (serviceIDs []string, err error) {
hash/dirhash/dirhash_test.go:10: Function 'TestDirHash' is too long (68 > 60) (funlen)
func TestDirHash(t *testing.T) {
container/service_test.go:91: Function 'TestStopService' is too long (85 > 60) (funlen)
func TestStopService(t *testing.T) {
process/filter_test.go:10: Function 'TestMatch' is too long (110 > 60) (funlen)
func TestMatch(t *testing.T) {
service/service_parameter_test.go:10: Function 'TestValidate' is too long (241 > 60) (funlen)
func TestValidate(t *testing.T) {
process/process_test.go:10: Function 'TestValidateProcess' is too long (206 > 60) (funlen)
func TestValidateProcess(t *testing.T) {
sdk/backend.go:33: Function 'initDefaultCosmosModules' is too long (100 > 60) (funlen)
func initDefaultCosmosModules(app *cosmos.AppFactory) {
hash/structhash/structhash_test.go:24: Function 'TestDump' is too long (123 > 60) (funlen)
func TestDump(t *testing.T) {
hash/structhash/structhash_test.go:151: Function 'TestTag' is too long (69 > 60) (funlen)
func TestTag(t *testing.T) {
sdk/execution/execution_listener_test.go:12: Function 'TestFilter' is too long (65 > 60) (funlen)
func TestFilter(t *testing.T) {
core/main.go:125: Function 'main' has too many statements (51 > 40) (funlen)
func main() {
```

We should fixes those function in our code but also exclude test files.

**Comment by krhubert** [comment]
We should deactivate that linter (same as gocyclo is) , because that linter is way far from being usable.

**Comment by NicolasMahe** [comment]
Can you elaborate and why it's not usable? same for gocyclo?
I think the default value are too low but it's a good practice to keep function small and without too much cycle.
There is exception of course. In this case, comment in code like `//nolint:gocyclo` should be used.

**Comment by krhubert** [comment]
> Can you elaborate and why it's not usable? same for gocyclo?

The best linters is one that runs out of the box and you don't have to configure them. This is not the case with len or gocyclo. To have an example for each one:

- (len) https://github.com/moby/moby/blob/master/client/interface.go - it looks nicer when all the methods are in oneline although the line is long.
- (gocyclo) https://github.com/mesg-foundation/engine/blob/dev/x/xnet/dnsclient.go#L11 - this function is not as complicated as gocyclo claims

Gocyclo is very quite simple tool and checking code complexity is very challenging task. The len is simple too, but still just too simple for a smart line counting. 

> In this case, comment in code like //nolint:gocyclo should be used.

I know how js code looks like with all those comments about exception for eslint. Sometimes there is more comment for all the rules then the code itself ;) So I don't want to end up with a lot of lint rules in the code which has to be maintained.

So to summarize on gocyclo example:

- the tool is too simple to be trusted so it won't help the reviewer
- putting `//nolint:gocyclo` in the code won't tell the reviewer if there is a pbm with the code because the linter is turn off
- if we want to avoid `//nolint:gocyclo` then limit has to be increased and in that case the results are useless

**Comment by antho1404** [comment]
I see the linter as help and cyclomatic complexity is a good indicator and if for some reason it really makes sense to have a fairly complex function I'm totally fine to ignore it with a nolint command.

I see this the same as codacy or these tools like that. It's good to have to help you improve the code quality if it makes sense but if it doesn't yes let's just ignore it for some specific cases.

**Comment by NicolasMahe** [comment]
I think we should use both tools but with higher limits and deactivate them when necessary.
I see this tools not as a requirements or proof that the code is good, but to prevent new devs to break the style and to help everyone to think twice if the lint breaks.
Check the uber's style guide: https://github.com/uber-go/guide/blob/master/style.md

Speaking of javascript, I really like [StandardJS](https://standardjs.com/) because it's specific rules on top of eslint. Do you know any similar configuration but for go? Something that a team already had all those discussion and came up with a pretty good lint rules.

**Comment by krhubert** [comment]
>I think we should use both tools but with higher limits and deactivate them when necessary.

So the 3rd option:


> Do you know any similar configuration but for go? 

Yes! We are using it https://github.com/golangci/golangci-lint/ right now. Which happened it has disabled by default linters such `gocyclo` or `lll`
